scio-avro/src/main/scala/com/spotify/scio/coders/avro/AvroCoders.scala

```diff
@@ -132,42 +123,22 @@ trait AvroCoders {
    * @return
    *   Coder[GenericRecord]
    */
-  // TODO: Use a coder that does not serialize the schema
   def avroGenericRecordCoder(schema: Schema): Coder[GenericRecord] =
-    Coder.beam(AvroCoder.of(schema))
+    avroCoder(GenericRecordDatumFactory, schema)
 
   // XXX: similar to GenericAvroSerializer
   def avroGenericRecordCoder: Coder[GenericRecord] =
     Coder.beam(new SlowGenericRecordCoder)
 
   implicit def avroSpecificRecordCoder[T <: SpecificRecord: ClassTag]: Coder[T] = {
-    val clazz = ScioUtil.classOf[T]
-    val schema = AvroCoders.schemaForClass(clazz).getOrElse {
-      val msg =
-        "Failed to create a coder for SpecificRecord because it is impossible to retrieve an " +
-          s"Avro schema by instantiating $clazz. Use only a concrete type implementing " +
-          s"SpecificRecord or use GenericRecord type in your transformations if a concrete " +
-          s"type is not known in compile time."
```

Comment:
I dropped this. Non standard avro classes should be handled using custom datum facory and coder